consensus/ibft/src/integration-test/java/tech/pegasys/pantheon/consensus/ibft/support/RoundSpecificNodeRoles.java

```diff
@@ -39,11 +48,31 @@ public ValidatorPeer getProposer() {
     return Collections.unmodifiableCollection(peers);
   }
 
-  public List<ValidatorPeer> getNonProposingPeers() {
```

Comment:
No. 

Providing this method invites the readability issues that I'm highlighting with PR. The solution your looking for is to change your approach ...rather then retrieving the list, can you imagine a different tact? perhaps one based in behaviour rather than direct access to a data structure?

Reply:
Alternatively, you merge you PR and I'll update the code when I get back around to coding later today ;)

Reply:
You're welcome to do that. I agree I don't like the current approach using the getter but until there is a solution to that, the getter is still needed.
Ideally i'd like to say something like assertNonProposingPeersReceived(proposalMsg)